data_catalog_backend/routes/resource_routes.py

```diff
@@ -0,0 +1,141 @@
+import logging
+import uuid
+
+from typing import List
+from fastapi import APIRouter, Depends, HTTPException
+
+from data_catalog_backend.models import SpatialExtent, Resource
+from data_catalog_backend.dependencies import get_license_service, get_resource_service, get_examples_service, \
+    get_provider_service, get_category_service
+from data_catalog_backend.schemas.resource import ResourcesResponse, ResourceRequest, ResourcesRequest, \
+    ResourceResponse
+from data_catalog_backend.services.category_service import CategoryService
+from data_catalog_backend.services.license_service import LicenseService
+from data_catalog_backend.services.provider_service import ProviderService
+from data_catalog_backend.services.resource_service import  ResourceService
+from data_catalog_backend.services.examples_service import ExamplesService
+
+router = APIRouter()
+
+@router.post(
+    "/resource",
+    summary="Add a resource to the database",
+    tags=["admin"],
+    response_model=ResourcesResponse)
+async def post_resource(
+        resource_req: ResourceRequest,
+        resource_service: ResourceService = Depends(get_resource_service),
+        license_service: LicenseService = Depends(get_license_service),
+        example_service: ExamplesService = Depends(get_examples_service),
+        provider_service: ProviderService = Depends(get_provider_service),
+        category_service: CategoryService = Depends(get_category_service)
+    ) -> ResourcesResponse:
+
+    license = license_service.create_or_find_license(resource_req.license)
+
+    providers = []
+    for provider_req in resource_req.providers:
+        provider = provider_service.create_or_find_provider(provider_req)
+        if not provider:
+            raise HTTPException(status_code=404, detail="Provider not found")
+        providers.append(provider)
+
+    categories = []
+    for category in resource_req.categories:
+        cat = category_service.create_or_find_category(category)
+        if not cat:
+            raise HTTPException(status_code=404, detail="Category not found")
+        categories.append(cat)
+
+    examples = []
+    if resource_req.examples:
+        examples = example_service.create_examples(resource_req.examples)
+
+    code_examples = []
+    if resource_req.code_examples:
+        code_examples = example_service.create_code_examples(resource_req.code_examples)
+
+    if not license:
+        raise HTTPException(status_code=404, detail="License not found")
+
+    spatial_extent_objects = []
+
+    if resource_req.spatial_extent is not None:
```

Comment:
This code could be shortened quite a bit, by being a bit more clever in the default-value in the resource_req. If the default-value is `[]` instead of None, we could skip the if check.

Also, if the pydantic-schemas and sqlalchemy-models have the same names, it is possible to shorthand the conversions. 

So the lines 63 - 73 could be shortened to the following:
`spatial_extent_objects = [SpatialExtent(**x.model_dump()) for x in resource_req.spatial_extent]`